packages/react-router/src/router.ts

```diff
@@ -1890,235 +1926,251 @@ export class Router<
             const validResolvedMatches = matches.slice(0, firstBadMatchIndex)
             const matchPromises: Array<Promise<any>> = []
 
-            validResolvedMatches.forEach((match, index) => {
-              const createValidateResolvedMatchPromise = async () => {
-                const parentMatchPromise = matchPromises[index - 1]
-                const route = this.looseRoutesById[match.routeId]!
-
-                const loaderContext: LoaderFnContext = {
-                  params: match.params,
-                  deps: match.loaderDeps,
-                  preload: !!preload,
-                  parentMatchPromise,
-                  abortController: match.abortController,
-                  context: match.context,
-                  location,
-                  navigate: (opts) =>
-                    this.navigate({ ...opts, _fromLocation: location }),
-                  cause: preload ? 'preload' : match.cause,
-                  route,
-                }
+            validResolvedMatches.forEach(
+              ({ id: matchId, routeId, params: p }, index) => {
+                const getMatch = () => getRouteMatch(this.state, matchId)!
+
+                const createValidateResolvedMatchPromise = async () => {
+                  const parentMatchPromise = matchPromises[index - 1]
+                  const route = this.looseRoutesById[routeId]!
+
+                  const {
+                    params,
+                    loaderDeps,
+                    abortController,
+                    context,
+                    cause,
+                  } = getMatch()
```

Comment:
Should this `getMatch()` call be moved so it gets invoked inside the `getLoaderContext()` function?
Functionally does it make a difference?

Reply:
Done